### ISSUE TITLE
Get the output of each layer

### ISSUE THREAD

**Issue by denru01** [ppwwyyxx/tensorpack#12]
Is there a way to get the output of each layer?

Thanks?

**Comment by ppwwyyxx** [comment]
Do you want to do this in inference time? Just add the names of those tensors to output names.

**Comment by denru01** [comment]
Thanks for your reply.
I would like to check the output of each layer in the DoReFa-Net example.
It uses this format to define the graph:
"logits = (LinearWrap(image)
                .Conv2D('conv0', 96, 12, stride=4, padding='VALID')
                .apply(activate)..."
How can I get the variable name of the output of each layer?
I have tried to use tf.all_variables(), but it returned an empty list.

**Comment by ppwwyyxx** [comment]
When model is written in this syntax-sugar, it'll be harder to find out names of those tensors.

What you can do is to split the expression and print them to find out the names:
```python
            logits = (LinearWrap(image)
                .Conv2D('conv0', 96, 12, stride=4, padding='VALID')
                .apply(activate)())
            print(logits)
            logits = (LinearWrap(logits)
           #.......
```

I also just added a convenient method in the latest commit. After a pull you can use the following:
```python
            logits = (LinearWrap(image)
                .Conv2D('conv0', 96, 12, stride=4, padding='VALID')
                .print()  # conv0/output
                .apply(activate)
                #.......
```
to find out the name of the tensors.

**Comment by ppwwyyxx** [comment]
Sorry but I found that the previous `.print()` feature only works in Python3 as `print` is a reserved keyword in Python2. I changed it do `.print_tensor()` instead.
